src/types/utilities.ts

```diff
@@ -6,3 +6,63 @@ export type FArguments<F> = F extends (...args: infer ARGS) => any
   : F extends new (...args: infer ARGS) => any
   ? ARGS
   : never;
+
+/**
+ * Pick a sparse type via a two-parameter path
+ *
+ * @example
+ *
+ * class Foo {
+ *   bar = {
+ *      baz: Promise.resolve(6)
+ *   }
+ *   a = 4;
+ *   b = { c: 'This is c'}
+ * }
+ *
+ * const pickTwoFoo: Pick2<Foo, 'b', 'c'> = new Foo();
+ * pickTwoFoo.b.c; // ☑️ will type-check
+ * pickTwoFoo.bar; // ❌ will NOT type-check
+ *
+ * const otherPickTwoFoo: Pick2<Foo, 'b', 'c'> = new Foo();
+ * otherPickTwoFoo.b; // ❌ will NOT type-check
+ * otherPickTwoFoo.bar.baz; // ☑️ will type-check
+ */
+export type Pick2<
+  OBJ,
+  A extends keyof OBJ = keyof OBJ,
+  B extends keyof OBJ[A] = keyof OBJ[A]
+> = {
+  [AA in A]: Pick<OBJ[AA], B>;
+};
+
+/**
+ * Pick a sparse type via a three-parameter path
+ *
+ * @example
+ *
+ * class Foo {
+ *   bar = {
+ *      baz: Promise.resolve(6)
+ *   }
+ *   a = 4;
+ *   b = {
+ *     c: {
+ *       d: 'This is d'
+ *     }
+ *   }
+ * }
+ *
+ * const pickThreeFoo: Pick3<Foo, 'b', 'c', 'd'> = new Foo();
+ * pickThreeFoo.b.c.d; // ☑️ will type-check
+ * pickThreeFoo.bar; // ❌ will NOT type-check
```

Comment:
Just a heads up the emojis cause js file to convert from UTF-8 to UTF-16 once its gone through the parser. Not sure about comments though.

Reply:
interesting. I'll remove the emoji, since they don't add a ton of value in this case.

Reply:
Resolved in #25